lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -504,14 +504,13 @@ parsePortDefList(OpAsmParser &parser, OperationState &result,
                  SmallVectorImpl<OpAsmParser::OperandType> &ports,
                  SmallVectorImpl<Type> &portTypes,
                  SmallVectorImpl<NamedAttrList> &portAttrs) {
-  auto parseElt = [&]() -> ParseResult {
+  auto parsePort = [&]() -> ParseResult {
     OpAsmParser::OperandType port;
     Type portType;
-    // FIXME: This will parse an empty series of commas?
-    if (failed(parser.parseOptionalRegionArgument(port)) ||
-        failed(parser.parseOptionalColon()) ||
-        failed(parser.parseType(portType)))
-      return success();
+    // Expect each port to have the form `%<ssa-name> : <type>`
+    if (failed(parser.parseRegionArgument(port)) ||
+        failed(parser.parseColon()) || failed(parser.parseType(portType)))
```

Comment:
Awesome, you can simplify this further to:

```
 if (parser.parseRegionArgument(port) ||
        parser.parseColon() || parser.parseType(portType))
```

Reply:
Good call. Fixed in ead85fe.